mmf/modules/embeddings.py

```diff
@@ -351,84 +336,138 @@ def forward(
+                position_embeddings_visual = position_embeddings_visual[
+                    :, : visual_embeddings.size(1), :
+                ]
+
+            position_embeddings_visual = (
+                position_embeddings_visual
+                + self.position_embeddings_visual(position_ids_visual)
+            )
+        else:
+            position_ids_visual = torch.zeros(
+                *visual_embeddings.size()[:-1], dtype=torch.long,
+                device=visual_embeddings.device
+            )
+            position_embeddings_visual = self.position_embeddings_visual(
+                position_ids_visual
+            )
 
-                # When fine-tuning the detector , the image_text_alignment is
-                # sometimes padded too long.
-                if position_embeddings_visual.size(1) != visual_embeddings.size(1):
-                    assert position_embeddings_visual.size(1) >= visual_embeddings.size(
-                        1
-                    )
-                    position_embeddings_visual = position_embeddings_visual[
-                        :, : visual_embeddings.size(1), :
-                    ]
+        return position_embeddings_visual
 
-                position_embeddings_visual = (
-                    position_embeddings_visual
-                    + self.position_embeddings_visual(position_ids_visual)
-                )
-            else:
-                position_ids_visual = torch.zeros(
-                    *visual_embeddings.size()[:-1],
-                    dtype=torch.long,
-                    device=visual_embeddings.device
-                )
-                position_embeddings_visual = self.position_embeddings_visual(
-                    position_ids_visual
-                )
+    def forward(
+        self,
+        input_ids,
+        token_type_ids=None,
+        visual_embeddings=None,
+        visual_embeddings_type=None,
+        position_embeddings_visual=None,
+        image_text_alignment=None,
+    ):
+        """
+        input_ids = [batch_size, sequence_length]
+        token_type_ids = [batch_size, sequence_length]
+        visual_embedding = [batch_size, image_feature_length, image_feature_dim]
+        image_text_alignment = [batch_size, image_feature_length, alignment_dim]
+        """
 
-            v_embeddings = (
-                visual_embeddings
-                + position_embeddings_visual
-                + token_type_embeddings_visual
+        # text embeddings
+        text_embeddings = self.encode_text(input_ids, token_type_ids=token_type_ids)
+
+        # visual embeddings
+        if visual_embeddings is not None:
+            v_embeddings = self.encode_image(
+                visual_embeddings,
+                visual_embeddings_type=visual_embeddings_type,
+                position_embeddings_visual=position_embeddings_visual,
+                image_text_alignment=image_text_alignment,
```

Comment:
Would *args and **kwargs need to be included here as well?

Reply:
And also where `get_position_embeddings_visual` and `encode_text` are called?

Reply:
no, they only need to be added to signature not to the actual call.

Reply:
Anything else we need to do for this ? :)

Reply:
Nope, once @vedanuj approves it internally, it should land automatically.